### PR TITLE
Tweak post dates and other styles in `PostTypeList` and `PostItem`

### PR DESCRIPTION
**Pull request by nylen**
This PR is part of a larger epic and behind the `posts/post-type-list` feature flag.  See p8F9tW-hL-p2.

It changes the `PostTypeList` display in the following ways:

- Uses full dates like "August 30, 2017 4:46 PM" for posts older than 1 week
- Removes the clock icon from the post date display

This PR renames `PostRelativeTime` to `PostTime` since the old name is no longer accurate, and makes the needed changes inside that component.

This changes the display of the `PostItem` component, which is also used in production to show the drafts list of the current (non-feature-flagged) `/posts` view.  Here is how this section of the screen changes as a result - I think the changes are acceptable, and preferable to maintaining branching logic inside the `PostTime` component:

| Before | After |
|--|--|
| 2017-08-30t17 19 39-0500 | 2017-08-30t17 21 35-0500 |

### `PostTypeList` before screenshot

2017-08-30t17 16 37-0500

### `PostTypeList` after screenshot

2017-08-30t17 07 19-0500

### To test

- Check out this branch
- Start Calypso with `ENABLE_FEATURES=posts/post-type-list npm start`
- Navigate to Blog Posts
- Verify that for recent post dates, the date is shown in relative terms like "2 days ago" (as before)
- Verify that for post dates 1 week ago or older, the date is shown in absolute terms like "August 30, 2017 4:46 PM"
- Verify that post dates no longer show a small clock icon on their left side.

### CODE REVIEW

**Comment by gwwar** [comment]
Some missed lint errors:

```
/home/ubuntu/wp-calypso/client/blocks/post-time/index.jsx
  4:1  error  React.PropTypes is deprecated since React 15.5.0, use the npm module prop-types instead  react/no-deprecated
  9:8  error  'Gridicon' is defined but never used                                                     no-unused-vars
```

File: client/blocks/post-time/index.jsx [hunk 1]
```diff
@@ -13,31 +13,40 @@ import Gridicon from 'gridicons';
...
+		if ( time.isBefore( moment().subtract( 7, 'days' ) ) ) {
+			// "August 30, 2017 4:46 PM" in English locale
+			timeDisplay = time.format( 'LLL' );
+		} else {
+			// "3 days ago" in English locale
+			timeDisplay = time.fromNow();
+		}
+	} else {
+		// Placeholder text: "a few seconds ago" in English locale
+		timeDisplay = moment().fromNow();
```
Comment (gwwar) [inline client/blocks/post-time/index.jsx]: If we set the default as the following, I believe we can remove this else block.
```
let timeDisplay = moment().fromNow();
```

File: client/my-sites/post-type-list/index.jsx [hunk 2]
```diff
@@ -26,7 +26,7 @@ import PostTypeListEmptyContent from './empty-content';
 /**
  * Constants
  */
-const DEFAULT_POST_ROW_HEIGHT = 86;
+const DEFAULT_POST_ROW_HEIGHT = 81;
```
Comment (gwwar) [inline client/my-sites/post-type-list/index.jsx]: Was this a bit too tall before?

**Comment by gwwar** [comment]
Somewhat unrelated, but it looks like the author alignment is a bit off:
screen shot 2017-08-30 at 5 04 48 pm

File: client/my-sites/post-type-list/index.jsx (hunk 2 above)
Comment (nylen) [reply client/my-sites/post-type-list/index.jsx]: The `PostItem` got shorter with the removal of the clock icon in this PR.  Also related to the author weirdness below, this was probably in line previously.

**Comment by nylen** [comment]
> Somewhat unrelated, but it looks like the author alignment is a bit off

Good catch, looks like I missed testing this PR with a multi-author site.  We can fix this up here or in https://github.com/Automattic/wp-calypso/pull/17549, whichever one is merged last.  I'll coordinate with my team to see what we want to do there.

Thanks for the review!

**Comment by nylen** [comment]
#17549 is merged; I'll update this PR accordingly.

**Comment by nylen** [comment]
This PR has become a bit more of a style cleanup after landing #17549 and #17550.  Results, in addition to the original date changes:

> 

>

File: client/blocks/post-time/index.jsx (hunk 1 above)
Comment (nylen) [reply client/blocks/post-time/index.jsx]: I wanted to keep all cases and calculations explicit here, so I moved this logic into a separate function in d826512873c70a9b6bdd53439e3f7e2d6fdcfba7.

**Comment by nylen** [comment]
Thanks for the review!

**Pull request merged**